### PR TITLE
add bulk-scan-processor client to get sas token

### PR DESCRIPTION
**Pull request by mdayican**
### Change description ###

add bulk-scan-processor client to get sas token

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+@AutoConfigureWireMock(port = 0)
+@ActiveProfiles("integration-test")
+@SpringBootTest(properties = "bulk-scan-processor-url=http://localhost:${wiremock.server.port}")
+public class BulkScanProcessorClientTest {
+
+    @Autowired
+    private BulkScanProcessorClient client;
+
+    @Test
+    public void should_return_sas_token_with_when_everything_is_ok_with_request() throws JsonProcessingException {
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java]: something missing after 'with'

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        // then
+        assertThat(sasResponse).isEqualToComparingFieldByField(expected);
+    }
+
+    @Test
+    public void should_throw_exception_when_requested_service_is_not_configured() throws Exception {
+
+        stubWithResponse("notFoundService", badRequest());
+
+        // when
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java]: missing `//given` and `//then`

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            FeignException.BadRequest.class
+        );
+
+        assertThat(exception.status()).isEqualTo(400);
+
+    }
+
+    private void stubWithResponse(String service, ResponseDefinitionBuilder builder) {
+        stubFor(get("/token/" + service).willReturn(builder));
+    }
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java]: imho it's clearer to get rid of this method it's just hides what's actually happening. You'll get same number of lines and one method to go to less.

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java (hunk 1 above)
Comment (mdayican) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java]: fix the name

File: src/main/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClient.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@FeignClient(
+    name = "bulk-scan-processor-client",
+    url = "${bulk-scan-processor-url}"
+)
+public interface BulkScanProcessorClient {
+
+    @GetMapping(value = "/token/{container}", consumes = APPLICATION_JSON_VALUE)
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClient.java]: Tests say the url part is `service`. Here it's `container`.
Not sure which one is correct, but should definitely be consistent.

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java (hunk 3 above)
Comment (mdayican) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java]: removed

File: src/main/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClient.java (hunk 4 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClient.java]: in processor it is service. changed toservice

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+        // when
+        SasTokenResponse sasResponse = client.getSasToken("sscs");
+
+        // then
+        assertThat(sasResponse).isEqualToComparingFieldByField(expected);
+    }
+
+    @Test
+    public void should_throw_exception_when_requested_service_is_not_configured() throws Exception {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java]: should it be not be 404 when service is not found ?

Comment (nitinprabhu) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java]: ohh my bad does bulk scan processor return 400 for invalid service ?

Comment (mdayican) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/clients/BulkScanProcessorClientTest.java]: yes, this test from processor test.

**Comment by rkondratowicz** [comment]
conflicts

**Pull request merged**